volatility3/framework/plugins/windows/psscan.py

```diff
@@ -269,39 +269,60 @@ def _generator(self):
             filter_func=pslist.PsList.create_pid_filter(self.config.get("pid", None)),
         ):
             file_output = "Disabled"
-            if self.config["dump"]:
-                # windows 10 objects (maybe others in the future) are already in virtual memory
-                if proc.vol.layer_name == kernel.layer_name:
-                    vproc = proc
+
+            # windows 10 objects (maybe others in the future) are already in virtual memory
+            # if the proc native_layer_name and layer_name match then it is in 'virtual' memory.
+            if proc.vol.layer_name == proc.vol.native_layer_name:
+                # proc is already in a virtual mem, so a new object is not needed. it means
+                # that if physical addresses are requested in the output then proc.vol.offset
+                # cannot be used because it will be virtual, so the mapping is needed.
+                vproc = proc
+                if self.config["physical"]:
+                    # the display should be physical addresses, so proc cannot be used. The
+                    # mappings are needed to find where it would be physically.
+                    _, _, offset, _, _ = list(
+                        memory.mapping(offset=proc.vol.offset, length=0)
+                    )[0]
                 else:
-                    try:
-                        vproc = self.virtual_process_from_physical(
-                            self.context,
-                            self.config["kernel"],
-                            proc,
-                        )
-                    except exceptions.PagedInvalidAddressException:
-                        vproc = None
+                    # the display should be virtual addresses, so proc can be used
+                    offset = proc.vol.offset
+
+                #  renderers.UnreadableValue()
+            else:
+                # proc is in virtual mem, so a new object needs to be creatd.
+                vproc = self.virtual_process_from_physical(
+                    self.context, self.config["kernel"], proc
+                )
+                if self.config["physical"]:
+                    # the display should be physical addresses, so proc can be used
+                    # as it is
+                    offset = proc.vol.offset
+                else:
+                    # the display should be virtual address, so vproc should be used
+                    # however virtual_process_from_physical is not always able to create
+                    # a vproc, in that case we need to display a UnreadableValue()
+                    if vproc is not None:
+                        offset = vproc.vol.offset
+                    else:
+                        offset = None
 
+            if self.config["dump"]:
```

Comment:
Ideally we'd guard for `if vproc is None:`.  If we hand it in, it will debug log but otherwise return None, so it's not catastrophic, but feels like an easy thing we can check (and catchall exception handlers aren't ideal, so that may go away at some point in the future)...